### DISCUSSION TITLE
Inconsistent display with x-show

### DISCUSSION THREAD

**Discussion by JeanDavidDaviet** [alpinejs/alpine#2966] [7. Bugs (archived)] [answered]
Hi thanks for your work on this library, it has always worked for me.
But I have an issue with a code I have where it seems the x-show attribute isn't consistent.

I have this piece of code : 

in the html template  : 
```html
<div x-data="test">
    <div x-show="true && false">hidden</div>
    <div x-show="Object.keys(websites).length === 0 && websitesLoading === false">shown</div>
    <div x-show="Object.keys(websites).length === 0">hidden</div>
    <div x-show="websitesLoading === false">shown</div>
</div>
``` 

in the javascript file : 
```javascript
document.addEventListener('alpine:init', () => {
    Alpine.data('test', () => {
        websites: [],
        websitesLoading: false,
        resultsLoading: false,
        async getWebsites(){
            this.websitesLoading = true;
            const response = await fetch(`/websites`);
            this.websitesLoading = false;
            if(response.status === 200){
                this.websites = await response.json();
            }
        }
    })
});
```

This gives the following result : 
![x-show](https://user-images.githubusercontent.com/1676202/170998618-fadcb457-eed1-4165-9b88-0a07317432c6.png)

As you can see the third div is hidden because the predicate `Object.keys(websites).length === 0` is false. And the predicate `websitesLoading === false` is true so the fourth div is correctly showed.
Whereas the second div, which should be shown only if the two predicated are true, is shown with only one of the two predicates resolving to true.

Is it an issue with the way Object.keys and the use of proxies work ? Or am I missing something ?
Sorry if this is an issue on my side, and thank for the great library 👍

**Comment by SimoTod** [comment, marked as answer]
I copied the code and none of the div are showing for me. The length property of an array object is not enumerable so you can't access it in that way. Object.keys(yourArray).length should always be undefined as far as I know.
